tests/andromede/test_electrolyzer_n_inputs.py

```diff
@@ -0,0 +1,301 @@
+    database.add_data("ez1", "alpha", ConstantData(0.7))
+
+    database.add_data("ep2", "p_max", ConstantData(100))
+    database.add_data("ep2", "cost", ConstantData(30))
+    database.add_data("ez2", "alpha", ConstantData(0.7))
+
+    database.add_data("gd", "demand", ConstantData(70))
+    database.add_data("gp", "p_max", ConstantData(10))
+    database.add_data("gp", "cost", ConstantData(40))
+
+    network = Network("test")
+    network.add_node(elec_node_1)
+    network.add_component(electric_prod_1)
+    network.add_component(electrolyzer1)
+    network.add_node(elec_node_2)
+    network.add_component(electric_prod_2)
+    network.add_component(electrolyzer2)
+    network.add_node(gaz_node)
+    network.add_component(gaz_prod)
+    network.add_component(gaz_demand)
+
+    network.connect(
+        PortRef(electric_prod_1, "balance_port"), PortRef(elec_node_1, "balance_port")
+    )
+    network.connect(
+        PortRef(elec_node_1, "balance_port"), PortRef(electrolyzer1, "FlowDI")
+    )
+    network.connect(PortRef(electrolyzer1, "FlowDO"), PortRef(gaz_node, "balance_port"))
+    network.connect(
+        PortRef(electric_prod_2, "balance_port"), PortRef(elec_node_2, "balance_port")
+    )
+    network.connect(
+        PortRef(elec_node_2, "balance_port"), PortRef(electrolyzer2, "FlowDI")
+    )
+    network.connect(PortRef(electrolyzer2, "FlowDO"), PortRef(gaz_node, "balance_port"))
+    network.connect(
+        PortRef(gaz_node, "balance_port"), PortRef(gaz_demand, "balance_port")
+    )
+    network.connect(
+        PortRef(gaz_prod, "balance_port"), PortRef(gaz_node, "balance_port")
+    )
+
+    scenarios = 1
+    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+    status = problem.solver.Solve()
+
+    assert status == problem.solver.OPTIMAL
+    assert problem.solver.Objective().Value() == 2971.4285714285716
+
+
+def test_electrolyzer_n_inputs_2():
+    """
+    Test with one electrolyzer that has two inputs
+    """
+
+    elec_node_1 = Node(model=NODE_BALANCE_MODEL, id="e1")
+    elec_node_2 = Node(model=NODE_BALANCE_MODEL, id="e2")
+    gaz_node = Node(model=NODE_BALANCE_MODEL, id="g")
+
+    electric_prod_1 = create_component(model=GENERATOR_MODEL, id="ep1")
+    electric_prod_2 = create_component(model=GENERATOR_MODEL, id="ep2")
+
+    gaz_prod = create_component(model=GENERATOR_MODEL, id="gp")
+    gaz_demand = create_component(model=DEMAND_MODEL, id="gd")
+
+    electrolyzer = create_component(model=TWO_INPUTS_CONVERTOR_MODEL, id="ez")
+
+    database = DataBase()
+    database.add_data("gd", "demand", ConstantData(70))
+    database.add_data("ep1", "p_max", ConstantData(100))
+    database.add_data("ep2", "p_max", ConstantData(100))
+    database.add_data("ep1", "cost", ConstantData(30))
+    database.add_data("ep2", "cost", ConstantData(30))
+    database.add_data("ez", "alpha1", ConstantData(0.7))
+    database.add_data("ez", "alpha2", ConstantData(0.5))
```

Comment:
You should reuse the same data between all tests of the same system to check that we indeed get the same solution in the different modellings as they should represent the same system